### DISCUSSION TITLE
generateStaticParams on dynamic route and throwing an error

### DISCUSSION THREAD

**Discussion by rwieruch** [vercel/next.js#62152] [App Router]
Reproduction:

Cache a dynamic page (e.g. `[ticketId]`) with `generateStaticParams`:

```
export async function generateStaticParams() {
  const tickets = await prisma.ticket.findMany();

  return tickets.map((ticket) => ({
    params: {
      ticketId: ticket.id,
    },
  }));
}
```

Fetch the `ticket` on this page with streaming:

```
import { prisma } from '@/services/prisma';

export const getTicket = async (id: string) => {
  return await prisma.ticket.findUniqueOrThrow({
    where: {
      id,
    },
  });
};
```

Visit a URL where there is no `ticket` and Prisma (or something else) throws an error.

----------

On development:

I see the error.tsx file showing up (and Next.js' `1 error` toast message).

On production, because the pages are not dynamic anymore:

I see a "Internal Server Error" by Next.

-----

Expected:

I am not sure what to expect. What's the best practice here? Do I need to adjust the data fetching this way?

```
import { prisma } from '@/services/prisma';

export const getTicket = async (id: string) => {
  try {
    const ticket = await prisma.ticket.findUniqueOrThrow({
      where: {
        id,
      },
    });
  
    return { ticket, error: null };
  } catch (error) {
    return { ticket: null, error };
  }
};
```

and handle the error myself on these pages? Or is there another best practice?

**Comment by gabriel-logan** [comment]
If you are using nextjs, possible causes, as you did not show the error so it is difficult.

       ticketId: ticket.id, possibly ID is a number so it needs to be converted to a string first, as this is how the return from generateStaticParams works, so do it

       ticketId: ticket.id.toString(),

second, when getting the data in the components, use destructuring

({params}) and not directly (id)

Take a look at my example, it will help you. I believe
https://github.com/gabriel-logan/generateStaticParams-nextjs14
